components/builder-originsrv/src/migrations/origin_projects.rs

```diff
@@ -446,5 +446,84 @@ pub fn migrate(migrator: &mut Migrator) -> SrvResult<()> {
                      END
                  $$ LANGUAGE plpgsql VOLATILE"#,
     )?;
+    migrator.migrate(
+        "originsrv",
+        r#"ALTER TABLE IF EXISTS origin_project_integrations
+            ADD COLUMN IF NOT EXISTS project_id bigint REFERENCES origin_projects(id) ON DELETE CASCADE NOT NULL,
+            ADD COLUMN IF NOT EXISTS integration_id bigint REFERENCES origin_integrations(id) ON DELETE CASCADE NOT NULL"#,
+    )?;
+    migrator.migrate(
+        "originsrv",
+        r#"INSERT INTO origin_project_integrations (project_id, integration_id)
+            SELECT op.id as project_id, oi.id as integrations_id
+            FROM origin_project_integrations opi
+            JOIN origin_projects op ON opi.name = op.package_name
+            JOIN origin_integrations as oi on opi.integration = oi.name
+            WHERE opi.project_id IS NULL
+            AND opi.integration_id IS NULL;"#,
+    )?;
+    migrator.migrate(
+        "originsrv",
+        r#"ALTER TABLE origin_project_integrations
+            DROP COLUMN IF EXISTS name,
```

Comment:
If we keep running migrations repeatedly, I think this is going to cause the previous step to fail.

Reply:
Also, what do you think about setting all the columns to `NOT NULL`, too?

Reply:
@christophermaier this is now guarded and I added in the null checks.

Reply:
![tenor-24283464](https://user-images.githubusercontent.com/207178/33457543-c4cf0778-d5f0-11e7-8728-13f1bb8a2b08.gif)